src/engine/source/schemf/src/schema.cpp

```diff
@@ -98,7 +98,26 @@ const Field& Schema::get(const DotPath& name) const
 
         if (it != name.cend() - 1)
         {
-
+            // Handle arrays e.g. "field/0"
+            // If the field is an array and the next part is the last one and a number, return a new field with the
+            // array type
+            if (entry->second.isArray() && (it + 1) == (name.cend() - 1))
```

Comment:
improve-desing: I notice that the first part of this block is repeated for both get() and has_field(), could you modulate it a little more?

Reply:
The little details makes it not so simple to be the same for both, and it its only used on those two functions.